### PR TITLE
core: add option to specify the chunk separator in `merge_message_runs`

### PR DESCRIPTION
**Pull request by CastaChick**
**Description:**
LLM will stop generating text even in the middle of a sentence if `finish_reason` is `length` (for OpenAI) or `stop_reason` is `max_tokens` (for Anthropic).
To obtain longer outputs from LLM, we should call the message generation API multiple times and merge the results into the text to circumvent the API's output token limit.
The extra line breaks forced by the `merge_message_runs` function when seamlessly merging messages can be annoying, so I added the option to specify the chunk separator.

**Issue:**
No corresponding issues.

**Dependencies:**
No dependencies required.

**Twitter handle:**
@hanama_chem
https://x.com/hanama_chem

### CODE REVIEW

File: libs/core/langchain_core/messages/utils.py [hunk 1]
```diff
@@ -425,6 +425,8 @@ def filter_messages(
 @_runnable_support
 def merge_message_runs(
     messages: Union[Iterable[MessageLikeRepresentation], PromptValue],
+    *,
+    with_newline_separator: bool = True,
```
Comment (baskaryan) [inline libs/core/langchain_core/messages/utils.py:429]: thoughts on making the arg:
```chunk_separator: str="\n"```

to give end user full control

Comment (CastaChick) [reply libs/core/langchain_core/messages/utils.py:429]: I think so, too. I'll make the changes. Thank you for your review 😄

Comment (CastaChick) [reply libs/core/langchain_core/messages/utils.py]: @baskaryan 
I have made the necessary changes. Could you please review them?

Comment (CastaChick) [reply libs/core/langchain_core/messages/utils.py]: @baskaryan Just a friendly reminder, could you please check my changes?

**Comment by CastaChick** [comment]
@baskaryan Apologies for the repeated reminders, but could you please review this?

**Comment by CastaChick** [comment]
@baskaryan I apologize for the repeated contact. Could you please review my pull request?

**Review by baskaryan — APPROVED** [review]

**Comment by baskaryan** [comment]
apologies for the delay, thank you for the improvement!

**Comment by CastaChick** [comment]
@baskaryan No problem at all. Thank you very much for the review. 
One of the CI jobs is failing due to an unknown error, is there anything I can do?

**Pull request merged**